CRM/Member/Form/MembershipRenewal.php

```diff
@@ -785,35 +790,20 @@ public function processMembership($contactID, $membershipTypeID, $is_test, $chan
       $membershipTypeID,
       $numRenewTerms
     );
-    $memParams = [
-      'membership_type_id' => $membershipTypeID,
+    $memParams = array_merge($memParams, [
       'end_date' => $dates['end_date'] ?? NULL,
-      'join_date' => $currentMembership['join_date'],
       'start_date' => $isMembershipCurrent ? $currentMembership['start_date'] : ($dates['start_date'] ?? NULL),
-      'id' => $currentMembership['id'],
-      'is_test' => $is_test,
```

Comment:
@eileenmcnaughton this doesn't seem to have made it into the `$memParams` nor the modified_id param ( tho the latter might be taken care of separately)

Reply:
@seamuslee001 yeah I mentioned that in the PR template - I think it's wrong - ie, what does it mean to renew a live membership in test mode? My best guess is it's a test payment in which case I think the test status of the membership should be unchanged

Reply:
so I'm thinking that it might impact on is_test is set on the corresponding contribution or not. I would note that is_test is set here https://github.com/civicrm/civicrm-core/blob/0c1281235ead0ba3767780c13d159199b26b2d3d/CRM/Member/Form/MembershipRenewal.php#L576 i.e. we set is_test for the contribution https://github.com/civicrm/civicrm-core/blob/683d3a35bde8ec4ee895a8e728659e7006d7edd1/CRM/Member/BAO/Membership.php#L2425 based on the original params passed into the create array https://github.com/civicrm/civicrm-core/blob/683d3a35bde8ec4ee895a8e728659e7006d7edd1/CRM/Member/BAO/Membership.php#L324

Reply:
@seamuslee001 there is a LOT of copy & paste - but if $isTest is coming from the membership we don't need to alter it in any way. if the membership is already is_test we are not changing that

Reply:
oh - & no - this code section doesn't alter the contribution - just the membership
Note modified-id is just moved to here https://github.com/civicrm/civicrm-core/pull/18365/files#diff-349ca8bcf1ef264bf3ca9acc2593aa54R757